gen/vsup_web

```diff
@@ -74,59 +74,84 @@ my $groups;  # $groups->{ID}->{NAME/USERS} = name/{uco} = 1;
 # AGGREGATE DATA
 #
 # FOR EACH USER
-foreach my $rData ($data->getChildElements) {
+foreach my $resourceId ($data->getResourceIds()) {
 
-	my %resourceAttributes = attributesToHash $rData->getAttributes;
-	my $relationType = $resourceAttributes{$A_R_RELATION_TYPE};
+	my $relationType = $data->getResourceAttributeValue(resource => $resourceId, attrName => $A_R_RELATION_TYPE);
 
-	my @groupsData = ($rData->getChildElements)[0]->getChildElements;
+	foreach my $groupId ($data->getGroupIdsForResource(resource => $resourceId)) {
 
-	foreach my $group (@groupsData) {
+		my $groupNasmespaceId = $data->getGroupAttributeValue(group => $groupId, attrName => $A_G_NS_ID);
+		my $groupNasmespaceName = $data->getGroupAttributeValue(group => $groupId, attrName => $A_G_NS_NAME);
+		my $groupCAOOrgan = $data->getGroupAttributeValue(group => $groupId, attrName => $A_G_CAO_ORGAN);
+		my $groupCAOFunkce = $data->getGroupAttributeValue(group => $groupId, attrName => $A_G_CAO_FUNKCE);
+		my $groupCAOFunkceId = $data->getGroupAttributeValue(group => $groupId, attrName => $A_G_CAO_FUNKCE_ID);
 
-		my %gAttributes = attributesToHash $group->getAttributes;
-
-		if (defined $gAttributes{$A_G_NS_ID} and defined $gAttributes{$A_G_NS_NAME}) {
-			$groups->{$gAttributes{$A_G_NS_ID}}->{"NAME"} = $gAttributes{$A_G_NS_NAME};
+		if (defined $groupNasmespaceId and defined $groupNasmespaceName) {
+			$groups->{$groupNasmespaceId}->{"NAME"} = $groupNasmespaceName;
 		}
 
-		my @membersData = ($group->getChildElements)[1]->getChildElements;
-		foreach my $member (@membersData) {
+		foreach my $memberId ($data->getMemberIdsForResourceAndGroup( resource => $resourceId, group => $groupId )) {
 
-			my %uAttributes = attributesToHash $member->getAttributes;
+			my $excluded = $data->getMemberAttributeValue( member => $memberId, attrName => $A_EXCLUDED );
```

Comment:
This should be ``getUserAttributeValue()``.

Reply:
Fixed